src/WinConsoleFile.cc

```diff
@@ -82,30 +86,34 @@ WinConsoleFile::WinConsoleFile(DWORD stdHandle)
       bold_(false),
       underline_(false),
       reverse_(false),
-      fg_(FOREGROUND_WHITE),
-      bg_(BACKGROUND_BLACK)
+      deffg_(FOREGROUND_WHITE),
+      defbg_(BACKGROUND_BLACK)
 {
   if (supportsColor()) {
     CONSOLE_SCREEN_BUFFER_INFO info;
     GetConsoleScreenBufferInfo(handle(), &info);
-    info.wAttributes &=
-        ~(COMMON_LVB_LEADING_BYTE | COMMON_LVB_TRAILING_BYTE |
-          COMMON_LVB_GRID_HORIZONTAL | COMMON_LVB_GRID_LVERTICAL |
-          COMMON_LVB_GRID_RVERTICAL | COMMON_LVB_REVERSE_VIDEO |
-          COMMON_LVB_UNDERSCORE);
-    fg_ = info.wAttributes &
-          ~(BACKGROUND_BLUE | BACKGROUND_GREEN | BACKGROUND_RED |
-            BACKGROUND_INTENSITY);
-    bg_ = info.wAttributes &
-          ~(FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED |
-            FOREGROUND_INTENSITY);
-    bg_ = (bg_ >> 4) & 0x0F;
+    deffg_ = info.wAttributes &
+             (FOREGROUND_BLUE | FOREGROUND_GREEN | FOREGROUND_RED);
+    defbg_ = info.wAttributes &
+             (BACKGROUND_BLUE | BACKGROUND_GREEN | BACKGROUND_RED);
```

Comment:
OK, so defbg_ is index to kBackground.  Then, this assignment is a bit confusing.  I think it is clearer to use local variable to store these temporary values.